python/graphstorm/dataloading/dataset.py

```diff
@@ -68,6 +68,40 @@ def prepare_batch_input(g, input_nodes,
                 for fname in feat_name], dim=1)
     return feat
 
+def prepare_batch_edge_input(g, input_edges,
+                             dev='cpu', feat_field='feat'):
+    """ Prepare minibatch edge input features
+
+    Note: The output is stored in dev.
+
+    Parameters
+    ----------
+    g: DGLGraph
+        The graph.
+    input_edges: dict of tensor
+        Input edges.
+    dev: th.device
+        Device to put output in.
+    feat_field: str or dict of list of str
+        Fields to extract features
+
+    Return:
+    -------
+    Dict of tensors.
+        If a node type has features, it will get node features.
+    """
+    feat = {}
+    for etypes, eid in input_edges.items():
+        feat_name = None if feat_field is None else \
+            [feat_field] if isinstance(feat_field, str) \
+            else feat_field[etypes] if etypes in feat_field else None
+
+        if feat_name is not None:
+            # concatenate multiple features together
+            feat[etypes] = th.cat([g.edges[etypes].data[fname][eid].to(dev) \
```

Comment:
we may consider to not use concatenation here.

Reply:
I want to make it a general function which can handle single feature per edge or multiple features per edge.

Reply:
my statement above is a general statement. there are cases we want to access each node data separately. but i don't think you need to address the problem in this PR.

Reply:
Agree.
That is a more complex use case.